src/Parser.js

```diff
@@ -67,7 +67,7 @@ export function parse (value, points, declarations, rules, rulesets) {
 						push(rulesets, ruleset(temporary, points, index, offset, rules, type, props = [], children = []))
 
 						if (character !== 59)
-							if (offset === 0)
+							if (offset === 0 || type === PAGE || type === FONT_FACE)
```

Comment:
I bet you might have some perf trick to make this faster 😅

Reply:
BTW offset is the length of the atrule so: 

```
@supports -> 8
@media -> 5
@font-page -> 9
@page -> 4
@keyframes -> 9
```

and you only need to test for `@media` and `@supports`.

```js
// the above code
else if (parse(value, points, props = [], atrules(offset, rules), children), sizeof(props))
// ... some function
function atrules (offset, rules) {
switch (offset) { case 8: case 5: return [''] }
return rules
}
```

Reply:
According to my tests it's not only about `rules` parameter, but also the last one (`rulesets`). In my working version  local `rulesets` are passed into `parse` but if we go with yours then local `children` are being passed down and this breaks stuff. 

Moreover - in my version first argument is `temporary` which is equal to a `@font-face` (in my test case) wherein your version it is `value` which is equal to `.user`. Not sure which one is correct, but it seems like passing context-like `@font-face` might be desired here.

The tip about testing 8 & 5 is sound, so maybe I should just tweak this if/else with that? Let me know what do you think, maybe what I've investigated here can shed some more light on a more desirable algo for you.

Reply:
What tests fail with this?

```js
else if (parse(value, points, props = [], offset === 6 || offset === 9 ? rules : [''], children) // ...
```

We are going to remove the first parameter being passed to parse because it is not being used at the moment.

Reply:
I assume that I was supposed to also remove the fix I've applied and only use what you have pasted in here. With that both currently failing tests on v4 branch are failing:
```
      at-rules:
      + expected - actual

      -@page{.user{size:A4 landscape;}}@document url(://www.w3.org/),url-prefix(//www.w3.org/),domain(mozilla.org),regexp("https:.*"){.user body{color:red;}}@viewport{.user{min-width:640px;max-width:800px;}}@counter-style list{.user{system:fixed;symbols:url();suffix:" ";}}@-moz-document url-prefix(){.selector{color:lime;}}@page{@bottom-right{{content:counter(pages);margin-right:1cm;}}.user{color:red;width:none;}}
      +@page{size:A4 landscape;}@document url(://www.w3.org/),url-prefix(//www.w3.org/),domain(mozilla.org),regexp("https:.*"){.user body{color:red;}}@viewport{min-width:640px;max-width:800px;}@counter-style list{system:fixed;symbols:url();suffix:" ";}@-moz-document url-prefix(){.user .selector{color:lime;}}@page{color:red;@bottom-right{content:counter(pages);margin-right:1cm;}width:none;}
      
       @font-face:
      + expected - actual

      -@font-face{.user{font-family:Pangolin;src:url('Pangolin-Regular.ttf') format('truetype');}}
      +@font-face{font-family:Pangolin;src:url('Pangolin-Regular.ttf') format('truetype');}
```

Reply:
> We are going to remove the first parameter being passed to parse because it is not being used at the moment.

As far as I can tell it's used by `ruleset` function (`parse` is passing it down there), maybe the usage can be removed though.

Reply:
I see, yes your method works better, though you can replace using TYPE with offset as a comparator. 

I'd turn the whole thing into a switch block and have `@supports` and `@media` and `@document` for one branch and `default` for the others since there are more of the "others" than the former.

The value can be removed as it looks like the reference passed to ruleset is not being used, i.e you can pass any value(i.e temporary) and it will still work.

Reply:
Scratch the last part: i noticed now that ruleset does use `value`, so that's important.

Reply:
Done, tests did not regress in comparison to the previous version of the fix 👍